apps/smart-contracts/core/test/PrePOMarket.test.ts

```diff
@@ -742,6 +765,111 @@ describe('=> prePOMarket', () => {
+
+      await prePOMarket.connect(user).redeem(longToRedeem, shortToRedeem)
+
+      expect(redeemHook.hook).to.not.be.called
+    })
+
+    it('approves fee for hook to use', async () => {
+      const amountMinted = await setupMarket()
+      const longToRedeem = amountMinted
+      const shortToRedeem = longToRedeem
+      const totalOwed = await calculateTotalOwed(longToRedeem, shortToRedeem, false)
+      expect(totalOwed).to.be.gt(0)
+      const redeemFee = calculateFee(totalOwed, await prePOMarket.getRedemptionFee())
+      expect(redeemFee).to.be.gt(0)
+
+      const tx = await prePOMarket.connect(user).redeem(longToRedeem, shortToRedeem)
+
+      expect(tx)
+        .to.emit(collateralToken, 'Approval')
+        .withArgs(prePOMarket.address, redeemHook.address, redeemFee)
+    })
+
+    it('sets approval back to 0', async () => {
+      const amountMinted = await setupMarket()
+      const longToRedeem = amountMinted
+      const shortToRedeem = longToRedeem
+      const totalOwed = await calculateTotalOwed(longToRedeem, shortToRedeem, false)
+      expect(totalOwed).to.be.gt(0)
+      const redeemFee = calculateFee(totalOwed, await prePOMarket.getRedemptionFee())
+      expect(redeemFee).to.be.gt(0)
+
+      const tx = await prePOMarket.connect(user).redeem(longToRedeem, shortToRedeem)
+
+      expect(tx)
+        .to.emit(collateralToken, 'Approval')
+        .withArgs(prePOMarket.address, redeemHook.address, redeemFee)
+      expect(tx)
+        .to.emit(collateralToken, 'Approval')
+        .withArgs(prePOMarket.address, redeemHook.address, 0)
+      expect(await collateralToken.allowance(prePOMarket.address, redeemHook.address)).to.eq(0)
+    })
+
+    it("doesn't approve fee if hook not set", async () => {
+      const amountMinted = await setupMarket()
+      const longToRedeem = amountMinted
+      const shortToRedeem = longToRedeem
+      await prePOMarket.connect(treasury).setRedeemHook(ZERO_ADDRESS)
+
+      const tx = await prePOMarket.connect(user).redeem(longToRedeem, shortToRedeem)
+
+      expect(tx).to.not.emit(collateralToken, 'Approval')
+      expect(await collateralToken.allowance(prePOMarket.address, redeemHook.address)).to.eq(0)
+    })
+
+    it('returns full amount owed if hook not set', async () => {
+      const amountMinted = await setupMarket()
+      const longToRedeem = amountMinted
+      const shortToRedeem = longToRedeem
+      await prePOMarket.connect(treasury).setRedeemHook(ZERO_ADDRESS)
+      const totalOwed = await calculateTotalOwed(longToRedeem, shortToRedeem, false)
+      expect(totalOwed).to.be.gt(0)
+      const redeemFee = calculateFee(totalOwed, await prePOMarket.getRedemptionFee())
+      expect(redeemFee).to.be.gt(0)
+
+      await prePOMarket.connect(user).redeem(longToRedeem, shortToRedeem)
+
+      expect(await collateralToken.balanceOf(user.address)).to.eq(totalOwed)
+    })
+
+    it('returns amount not spent by hook', async () => {
```

Comment:
```suggestion
    it('sends full collateral amount if hook takes no fee', async () => {
```

and also add test for when the hook takes part of the fee + all of the fee (if this is not possible now, we should leave a TODO to test it later another way)